packages/@lwc/integration-karma/test/regression/lifecycle/index.spec.js

```diff
@@ -0,0 +1,165 @@
+import { createElement } from 'lwc';
+import { extractDataIds } from 'test-utils';
+
+import NativeSloterSyntheticSlotee from 'x/nativeSloterSyntheticSlotee';
+import SyntheticSloterSyntheticSlotee from 'x/syntheticSloterSyntheticSlotee';
+import NativeSloterNativeSlotee from 'x/nativeSloterNativeSlotee';
+
+describe('W-15904769', () => {
+    beforeEach(() => {
+        window.timingBuffer = [];
+    });
+
+    afterEach(() => delete window.timingBuffer);
+
+    const assertSlottedChildrenLength = (elm, expected) => {
+        const actual = extractDataIds(elm);
+        expect(Object.keys(actual).filter((node) => !node.includes('.shadowRoot')).length).toBe(
+            expected
+        );
+    };
+
+    if (!process.env.NATIVE_SHADOW) {
+        it('native sloter + synthetic slotee disconnects root and subtree components but does not reconnect synthetic subtree', async () => {
+            const elm = createElement('x-native-sloter-synthetic-slotee', {
+                is: NativeSloterSyntheticSlotee,
+            });
+            document.body.appendChild(elm);
+            await Promise.resolve();
+
+            assertSlottedChildrenLength(elm, 3);
+            expect(window.timingBuffer.length).toBe(3);
+            expect(window.timingBuffer[0]).toBe('x-use-api-version-61-sloter: connectedCallback');
+            expect(window.timingBuffer[1]).toBe('x-use-api-version-61-sloter: connectedCallback');
+            expect(window.timingBuffer[2]).toBe('x-use-api-version-60-slotee: connectedCallback');
+
+            // Hide spans which reorders the children and causes x-use-api-version-61-sloter to be re-inserted in the DOM
+            elm.toggleSpans();
+            await Promise.resolve();
+            assertSlottedChildrenLength(elm, 1);
+            expect(window.timingBuffer.length).toBe(6);
+            expect(window.timingBuffer[3]).toBe(
+                'x-use-api-version-61-sloter: disconnectedCallback'
+            );
+            // Notice x-use-api-version-60-slotee disconnectedCallback is fired but not its connectedCallback
+            expect(window.timingBuffer[4]).toBe(
+                'x-use-api-version-60-slotee: disconnectedCallback'
+            );
+            expect(window.timingBuffer[5]).toBe('x-use-api-version-61-sloter: connectedCallback');
```

Comment:
When the sloter is moved, the browser will fire the [native `disconnectedCallback` ](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-dom/src/custom-elements/create-custom-element.ts#L55-L64)which will disconnect the entire subtree.

The browser will then call native `connectedCallback` which runs through rehydration > VDOM diffing to determine if it should unmount/mount/patch any of the elements.

Since the sloter's subtree structure hasn't changed (slotee hasn't changed), the engine determines there's no need to do anything and moves on.

The browser will then call the slotee's native `connectedCallback` but since it's using synthetic lifecycle (`apiVersion = 60` ) it will skip invoking.

We're left with the subtree having `disconnectedCallback` invoked but not a follow up `connectedCallback`.

We're observing 2 issues with this:

1. Customer are relying on these lifecycle methods to set state in their component.
2. We're seeing wire adapters disconnecting and not being reconnected.